ratschlab_common/io/sparse_df.py

```diff
@@ -0,0 +1,122 @@
+import pandas as pd
+import numpy as np
+from pandas.api.types import is_sparse
+from scipy import sparse
+import tables
+
+'''
+This module exposes two functions: to_hdf and read_hdf, which usage
+is to read and write sparse Pandas DataFrames into and from hdf file.
+'''
+
+
+def read_hdf(fpath):
+    '''
+    read a hdf file into a sparse Pandas DataFrame that was previously saved with the to_hdf function of this module.
+
+    :param fpath: location of the hdf file
+    :return: an instance of Pandas DataFrame
+    '''
+    f = tables.open_file(fpath, mode='r')
+    sparse_group = f.get_node("/data/sparse")
+    sparse_m = _read_sparse_m_coo(sparse_group)
+    s_col_names = f.get_node("/data/sparse/col_names")
+    non_sparse = pd.read_hdf(fpath, key="/data/non_sparse")
+    sparse = pd.DataFrame.sparse.from_spmatrix(sparse_m, columns=s_col_names)
+    ret_df = pd.concat([sparse, non_sparse], axis=1)
```

Comment:
Round tripping won't work if you have a dataframe like [sparse_col1, dense_col, sparse_col2]. It would get reconstructed as [sparse_col1, sparse_col2, dense_col]. Could you save all the column names and then do something like `ret_df[orig_cols]` s.t. the order is maintained overall?
Otherwise, users get surprised not seeing the same order of columns.